### DISCUSSION TITLE
Configuring and using different hostnames on iOS and Android

### DISCUSSION THREAD

**Discussion by tokkoro** [ionic-team/capacitor#4173] [General]
Hi, we are building our Angular single page app with Capacitor and we are using two different hostnames for the platforms. Is there a way to automate changing them depending on current platform I'm building for? Could I have it setup somewhere else than capacitor.config.json that as far as I know can only have one hostname at a time.

So, I'm using it like this:

Configures for android:

```
server: {
	hostname: "myappname-capacitor",
	hostname-ios: "app.mayappname.com"
}
```

and for iOS:

```
server: {
	hostname-android: "myappname-capacitor",
	hostname: "app.mayappname.com"
}
```

Is there a better way to do this? I also have the same web app hosted at app._myappname_.com and all API calls for the server goes under app._myappname_.com/api/*

(this is my first post on this forum, so please guide me if necessary).

**Comment by Schwankenson** [comment]
Did it by getting the command line parameter with `process.argv'. This is ugly but it works. Better solutions are welcome

    import { CapacitorConfig } from '@capacitor/cli';

    const config: CapacitorConfig = {
        (...)
    };

    // Overrides for Android
    if (process.argv[3] === "android") {
    config["server"]["hostname"] = "android.whatever.de"
    }

    export default config;

**Comment by Wilt** [comment]
You can use a node environment variable to set specifics in your config:

Check [the Capacitor documentation for an example here...](https://capacitorjs.com/docs/guides/environment-specific-configurations#dynamically-build-for-different-environments)

**Comment by Wilt** [comment]
I also found another neat trick using [the node command line arguments](https://stackoverflow.com/a/22214067/1697459)  mentioned [here in this post on the Ionic Forum](https://forum.ionicframework.com/t/how-to-autofill-password-on-ios/199744/58).

    // Third argument in node process should hold the platform
    const platform: 'ios'|'android' = process.argv[3];

Now you can use this inside your config.

    switch (platform) {
      case 'ios':
        // Do ios specific configuration
        break;
      case 'android':
        // Do android specific configuration
        break;
      default:
        throw Error('Unexpected third command line argument ${platform} in node command');
    }

Realized this is actually the same as @Schwankenson suggested, but I'll leave it since it seems like this a bit more readable/understandable and bulletproof...
